### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by kens-code**
Added the $help command.
Added the $my100stats command. This command mirrors the current $my100status command.
Fixed a looping glitch with the $my100status command.
Removed the $botstats command.

Enchanced the bots command checking to include a breakdown of the actual command and any associated parameters for the command. 

Reorganized folder structure to include bot, helper and messages directories.
Renamed JavaScript files in the /bot/commands directory to reflect the name of the actual.

Created a "commands.process" function that will check for an issues command and break that command down into the acutal command and any associated parameters.
Created a "commands.execute" function that will check for a valid command issued to the bot.
Created a "file.read" function to read content from the local file system.
Created a "messages.format" function that can load local content from the /messages/templates directory and dynamic replace values in that file with variables passed via the function's parameters.

Added usage checking for both the $my100status, $my100stats and $stats commands.

Improved the $test command by adding a check to make sure it is being running by the test bot instead of the live bot.

Various other code optimization.

### CODE REVIEW

File: bot.js [hunk 1]
```diff
@@ -1,186 +1,111 @@
...
-// Authorization
 const authDetails = require('./auth.json');
+const bot = new Discord.Client({ autoReconnect: true });
+const commands = require('./bot/commands.js');
 
-const prefix = '$';
-const adminPrefix = '-';
+// When the bot shuts down for whatever reason we post a msg in bottestchannel
+// to keep a log and notify bot admins.
+bot.on('disconnect', msg =>
```
Comment (jturbo26) [inline bot.js]: Is there a reason you moved this to the top?

I think we should keep these in some logical order. Maybe by the order the bot uses them or the frequency in which the bot uses them? Idea below:

1. bot.on(ready
2. bot.on(msg
3. bot.on(debug
4. bot.on(error
5. bot.on(disconnect

File: bot/commands.js [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+            var parameters;
+
+            data = msg.content.match(/(['"])((?:\\\1|.)+?)\1|([^\s"']+)/g);
+
+            command = data[0];
+
+            data.splice(0, 1);
+
+            parameters = data;
```
Comment (jturbo26) [inline bot/commands.js]: It doesn't look like parameters is actually ever used. Can this be removed?

File: bot/commands.js [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+            if (command.startsWith(adminPrefix) || command.startsWith(userPrefix))
```
Comment (jturbo26) [inline bot/commands.js]: Why have the double conditional? If it's true above, it should be true here too. Does it have to do with reversing the prefix order?

File: bot/commands/commands.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+﻿const messages = require('../../messages/format.js');
+
+const commands = (msg) =>
```
Comment (jturbo26) [inline bot/commands/commands.js:3]: With only one parameter parens aren't required here.

File: bot/commands/my100status.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            {
+                msg.reply('Sorry, no user was found by that name. Please try again with a different username.');
+			}
+            else
+            {
+                msg.reply(gamer.gamertag + ' has ' + gamer.activity_score + ' activity points and ' + gamer.karmas_count + ' karma on the site!');
+			}
+		}
+        else
+        {
```
Comment (jturbo26) [inline bot/commands/my100status.js]: This else does nothing and indentation is jacked. We should either have an error handler to handle not getting 200 from the server or just remove the else all together prob.

File: messages/format.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+const fs = require('fs');
+const path = require('path');
+const file = require('../helper/file.js');
+
+module.exports =
+{
+    format: function (fileName, ...replacements)
```
Comment (jturbo26) [inline messages/format.js]: Arrow function please

File: messages/format.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        var filePath = path.join(__dirname, "templates", fileName);
```
Comment (jturbo26) [inline messages/format.js]: Change var to const if it's a constant and let if it will be re-assigned please

**Review by jturbo26 — CHANGES_REQUESTED** [review]
I made some inline comments on specific lines I noticed things on. 

Overall I'd like to see the following please:
-Change all functions to arrow functions
-Remove all var and change to const or let depending on what the variable is.
-Convert all double quotes to single to match current code. (eslint is freaking on your PR)
-Since you went with spaces (needed to be done) can you make sure everything is spaces throughout. There are some loose tabs still.

We will have to chat so you can explain the text templates to me. I like the idea, but I don't totally understand how it's working.

The only thing that bothers me is you putting the opening bracket on the line following a function or conditional. I think that's just a personal choice thing though and I can come around to having it that way.

Other than that, thank you so much for everything you did here. With someone as experienced as you we can do some great things going forward!

File: bot/commands.js [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+                            // TODO: Create multigame lookup
+                            // Example: $stats overwatch [BattleTag] or $stats ow [BattleTag]
+
+                            help(msg, userPrefix + parameters[0], "Shows you a user's Overwatch quick play statistics.", "[BattleTag]", "[BattleTag] The Battle.net Tag of the user you would like to lookup.");
+
+                            break;
+
+                        case (parameters[0] === "test"):
+
+                            if (bot.user.tag == testbotjoe26Tag)
```
Comment (jturbo26) [inline bot/commands.js]: === here instead of ==

File: bot/commands.js [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+                default:
+
+                    help(msg, command, "Shows you a user's activity points and karma from the100.io site.", "[BattleTag]", "[BattleTag] The Battle.net Tag of the user you would like to lookup.");
+            }
+
+            break;
+
+        case (command === userPrefix + "playingnow"): // $playingnow
+
+            playingnow(msg)
```
Comment (jturbo26) [inline bot/commands.js]: missing ;

File: bot/commands.js [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+                    fields: [
+                        {
+                            name: command + ' not found',
+                            value: '```yaml\n' + `Type $commands to display a list of available commands` + '```'
+                        }
+                    ]
+                }
+            });
+    }
+}
```
Comment (jturbo26) [inline bot/commands.js]: missing ;

File: bot.js (hunk 1 above)
Comment (kens-code) [reply bot.js]: I placed the event handlers in alphabetical order as a systematic approach to speed up the process of finding a specific function as the number of event handlers continues to grow.

File: bot/commands.js (hunk 2 above)
Comment (kens-code) [reply bot/commands.js]: Removed both the local var parameters and the unused line parameters = data as this was used during testing and is no longer needed.

Comment (kens-code) [reply bot/commands.js]: Parameter has been updated to be used in the function call.

File: bot/commands.js (hunk 3 above)
Comment (kens-code) [reply bot/commands.js]: Removed duplicate check.

File: bot.js (hunk 1 above)
Comment (kens-code) [reply bot.js]: Rearranged event handlers as follows:

1. ready
2. message
3. guildMemberAdd
4. guildMemberRemove
5. disconnect
6. debug
7. warn
8. error

File: bot/commands/commands.js (hunk 4 above)
Comment (kens-code) [reply bot/commands/commands.js:3]: This is true but it also does not affect the execution of the execution of the code itself but makes it easier to declare no parameter or multiple parameters if needed.

() => { ... } // no parameter
(x) => { ... } // one parameter, an identifier
(x, y) => { ... } // several parameters

File: bot/commands/my100status.js (hunk 5 above)
Comment (kens-code) [reply bot/commands/my100status.js]: This was also used during testing and the else statement has been removed.

File: bot/commands.js [hunk 11]
```diff
@@ -0,0 +1,236 @@
...
+
+        case (command === userPrefix + 'games'): // $games
+
+            games(msg);
+
+            break;
+
+        case (command === userPrefix + 'help'): // $help
+
+            parameters[0] = parameters[0].replace(/\$/g, "");
```
Comment (kens-code) [inline bot/commands.js]: Found a bug while testing.

When issuing the $help command with no parameters, the parameters variable is undefined and thus caused an error to occur when the replace function was called.

Comment (kens-code) [reply bot/commands.js]: This has been fixed.

File: messages/format.js (hunk 6 above)
Comment (kens-code) [reply messages/format.js]: Replaced with arrow function

File: messages/format.js (hunk 7 above)
Comment (kens-code) [reply messages/format.js]: const and let statements used

File: bot/commands.js (hunk 8 above)
Comment (kens-code) [reply bot/commands.js]: This has been updated.

File: bot/commands.js (hunk 9 above)
Comment (kens-code) [reply bot/commands.js]: This has been updated.

File: bot/commands.js (hunk 10 above)
Comment (kens-code) [reply bot/commands.js]: That's strange. In my source, the semicolon is there on that line.

File: bot.js [hunk 12]
```diff
@@ -1,186 +1,110 @@
...
+        4. Check out ${bot.channels.find('name', 'use_enslaved_omnics_here')} and use the $games command to see what's on the schedule.
+
+        5. Read ${bot.channels.find('name', 'rules-missionstatement')} to learn more.
+
+        That's it. If you have any questions, please let a Core Member or Moderator know.
+		        `
+        }
+    });
+
+    const generalChannel = guildMember.guild.channels.find('name', 'general');
```
Comment (jturbo26) [inline bot.js]: This will need to change to "l-----------general-----------l" <---copy exactly inside quotes

**Review by jturbo26 — CHANGES_REQUESTED** [review]
Everything looks good, just need to change the general channel.

**Review by jturbo26 — APPROVED** [review]

**Pull request merged**